hashicorp/web-unified-docs
# content versions endpoint

# Changes

- Refactor current behaviour for this endpoint, which is to return **_all_** versions.
  - Added function to traverse nav data and return version(s) if file exists in matching product repo
  - Added vitest and tests for the content versions endpoint


- [Asana task](https://app.asana.com/0/1207899860738460/1208471953509347/f)

# Testing

1. Make a request to the endpoint in this PR:
```
curl https://web-presence-experimental-docs-git-heat-featco-2ba537-hashicorp.vercel.app/api/content-versions\?product\=terraform-cdk\&fullPath\=doc%23cdktf%2Fapi-reference%2Fpython
```

The response should look like this:
```
{"versions":["v0.12.x","v0.13.x","v0.14.x","v0.15.x","v0.16.x","v0.17.x","v0.18.x","v0.19.x","v0.20.x"]}%
```

2. Make a request to the current Content API:
```
curl https://content.hashicorp.com/api/content-versions\?product\=terraform-cdk\&fullPath\=doc%23cdktf%2Fapi-reference%2Fpython
```
The response should look like this:
```
{"versions":["dev-portal","v0.12.x","v0.13.x","v0.14.x","v0.15.x","v0.16.x","v0.17.x","v0.18.x","v0.19.x","v0.20.x"]}%    
```

The response returned by the endpoint modified in this PR should match the current Content API response (**except for `"dev-portal"`** as this version is not used anywhere in dev-portal AFAIK).
